### PR TITLE
Fix Swift builds

### LINKED ISSUE #231: Swift builds are broken on macOS
When building a simple CLI project on macOS, we currently get:

```
> xcodebuild -scheme ldk-node-test-cli-main build
Command line invocation:
    /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -scheme ldk-node-test-cli-main build

User defaults from command line:
    IDEPackageSupportUseBuiltinSCM = YES

Resolve Package Graph


Resolved source packages:
  ldk-node: /Users/USER/workspace/ldk-node/bindings/swift @ local

--- xcodebuild: WARNING: Using the first of multiple matching destinations:
{ platform:macOS, arch:arm64, id:00006001-000A69CE3C04801E }
{ platform:macOS, arch:x86_64, id:00006001-000A69CE3C04801E }
{ platform:macOS, name:Any Mac }
Prepare packages

ComputeTargetDependencyGraph
note: Building targets in dependency order
note: Target dependency graph (3 targets)
    Target 'ldk-node-test-cli-main' in project 'ldk-node-test-cli-main'
        ➜ Explicit dependency on target 'LDKNode' in project 'ldk-node'
    Target 'LDKNode' in project 'ldk-node'
        ➜ Explicit dependency on target 'LDKNode' in project 'ldk-node'
    Target 'LDKNode' in project 'ldk-node' (no dependencies)

GatherProvisioningInputs

CreateBuildDescription
Build description signature: f11216130748366c04924d9ecc99fc3c
Build description path: /Users/USER/Library/Developer/Xcode/DerivedData/ldk-node-test-cli-main-bthhyqunlbsecmebparzdfcqmiry/Build/Intermediates.noindex/XCBuildData/f11216130748366c04924d9ecc99fc3c.xcbuilddata

ClangStatCache /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang-stat-cache /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk /Users/USER/Library/Developer/Xcode/DerivedData/SDKStatCaches.noindex/macosx14.2-23C53-df0db8920d7ae99241a1bc0f08d2dced.sdkstatcache
    cd /tmp/ldk-node-test-cli-main/ldk-node-test-cli-main.xcodeproj
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang-stat-cache /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk -o /Users/USER/Library/Developer/Xcode/DerivedData/SDKStatCaches.noindex/macosx14.2-23C53-df0db8920d7ae99241a1bc0f08d2dced.sdkstatcache

ProcessProductPackaging "" /Users/USER/Library/Developer/Xcode/DerivedData/ldk-node-test-cli-main-bthhyqunlbsecmebparzdfcqmiry/Build/Intermediates.noindex/ldk-node-test-cli-main.build/Debug/ldk-node-test-cli-main.build/ldk-node-test-cli-main.xcent (in target 'ldk-node-test-cli-main' from project 'ldk-node-test-cli-main')
    cd /tmp/ldk-node-test-cli-main

    Entitlements:

    {
    "com.apple.application-identifier" = "";
    "com.apple.security.get-task-allow" = 1;
}

    builtin-productPackagingUtility -entitlements -format xml -o /Users/USER/Library/Developer/Xcode/DerivedData/ldk-node-test-cli-main-bthhyqunlbsecmebparzdfcqmiry/Build/Intermediates.noindex/ldk-node-test-cli-main.build/Debug/ldk-node-test-cli-main.build/ldk-node-test-cli-main.xcent

ProcessProductPackagingDER /Users/USER/Library/Developer/Xcode/DerivedData/ldk-node-test-cli-main-bthhyqunlbsecmebparzdfcqmiry/Build/Intermediates.noindex/ldk-node-test-cli-main.build/Debug/ldk-node-test-cli-main.build/ldk-node-test-cli-main.xcent /Users/USER/Library/Developer/Xcode/DerivedData/ldk-node-test-cli-main-bthhyqunlbsecmebparzdfcqmiry/Build/Intermediates.noindex/ldk-node-test-cli-main.build/Debug/ldk-node-test-cli-main.build/ldk-node-test-cli-main.xcent.der (in target 'ldk-node-test-cli-main' from project 'ldk-node-test-cli-main')
    cd /tmp/ldk-node-test-cli-main
    /usr/bin/derq query -f xml -i /Users/USER/Library/Developer/Xcode/DerivedData/ldk-node-test-cli-main-bthhyqunlbsecmebparzdfcqmiry/Build/Intermediates.noindex/ldk-node-test-cli-main.build/Debug/ldk-node-test-cli-main.build/ldk-node-test-cli-main.xcent -o /Users/USER/Library/Developer/Xcode/DerivedData/ldk-node-test-cli-main-bthhyqunlbsecmebparzdfcqmiry/Build/Intermediates.noindex/ldk-node-test-cli-main.build/Debug/ldk-node-test-cli-main.build/ldk-node-test-cli-main.xcent.der --raw

Ld /Users/USER/Library/Developer/Xcode/DerivedData/ldk-node-test-cli-main-bthhyqunlbsecmebparzdfcqmiry/Build/Products/Debug/ldk-node-test-cli-main normal (in target 'ldk-node-test-cli-main' from project 'ldk-node-test-cli-main')
    cd /tmp/ldk-node-test-cli-main
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -Xlinker -reproducible -target arm64-apple-macos14.2 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk -O0 -L/Users/USER/Library/Developer/Xcode/DerivedData/ldk-node-test-cli-main-bthhyqunlbsecmebparzdfcqmiry/Build/Intermediates.noindex/EagerLinkingTBDs/Debug -L/Users/USER/Library/Developer/Xcode/DerivedData/ldk-node-test-cli-main-bthhyqunlbsecmebparzdfcqmiry/Build/Products/Debug -F/Users/USER/Library/Developer/Xcode/DerivedData/ldk-node-test-cli-main-bthhyqunlbsecmebparzdfcqmiry/Build/Intermediates.noindex/EagerLinkingTBDs/Debug -F/Users/USER/Library/Developer/Xcode/DerivedData/ldk-node-test-cli-main-bthhyqunlbsecmebparzdfcqmiry/Build/Products/Debug/PackageFrameworks -F/Users/USER/Library/Developer/Xcode/DerivedData/ldk-node-test-cli-main-bthhyqunlbsecmebparzdfcqmiry/Build/Products/Debug -filelist /Users/USER/Library/Developer/Xcode/DerivedData/ldk-node-test-cli-main-bthhyqunlbsecmebparzdfcqmiry/Build/Intermediates.noindex/ldk-node-test-cli-main.build/Debug/ldk-node-test-cli-main.build/Objects-normal/arm64/ldk-node-test-cli-main.LinkFileList -Xlinker -rpath -Xlinker /Users/USER/Library/Developer/Xcode/DerivedData/ldk-node-test-cli-main-bthhyqunlbsecmebparzdfcqmiry/Build/Products/Debug/PackageFrameworks -Xlinker -object_path_lto -Xlinker /Users/USER/Library/Developer/Xcode/DerivedData/ldk-node-test-cli-main-bthhyqunlbsecmebparzdfcqmiry/Build/Intermediates.noindex/ldk-node-test-cli-main.build/Debug/ldk-node-test-cli-main.build/Objects-normal/arm64/ldk-node-test-cli-main_lto.o -Xlinker -export_dynamic -Xlinker -no_deduplicate -fobjc-link-runtime -L/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/macosx -L/usr/lib/swift -Xlinker -add_ast_path -Xlinker /Users/USER/Library/Developer/Xcode/DerivedData/ldk-node-test-cli-main-bthhyqunlbsecmebparzdfcqmiry/Build/Intermediates.noindex/ldk-node-test-cli-main.build/Debug/ldk-node-test-cli-main.build/Objects-normal/arm64/ldk_node_test_cli_main.swiftmodule -Wl,-no_warn_duplicate_libraries -framework LDKNodeFFI -framework LDKNodeFFI -Xlinker -no_adhoc_codesign -Xlinker -dependency_info -Xlinker /Users/USER/Library/Developer/Xcode/DerivedData/ldk-node-test-cli-main-bthhyqunlbsecmebparzdfcqmiry/Build/Intermediates.noindex/ldk-node-test-cli-main.build/Debug/ldk-node-test-cli-main.build/Objects-normal/arm64/ldk-node-test-cli-main_dependency_info.dat -o /Users/USER/Library/Developer/Xcode/DerivedData/ldk-node-test-cli-main-bthhyqunlbsecmebparzdfcqmiry/Build/Products/Debug/ldk-node-test-cli-main -Xlinker -add_ast_path -Xlinker /Users/USER/Library/Developer/Xcode/DerivedData/ldk-node-test-cli-main-bthhyqunlbsecmebparzdfcqmiry/Build/Intermediates.noindex/ldk-node.build/Debug/LDKNode.build/Objects-normal/arm64/LDKNode.swiftmodule
ld: Undefined symbols:
  _SCDynamicStoreCopyProxies, referenced from:
      system_configuration::dynamic_store::SCDynamicStore::get_proxies::hccd33d1a0964091e in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _SCDynamicStoreCreateWithOptions, referenced from:
      system_configuration::dynamic_store::SCDynamicStore::create::h61b7511cd088ff72 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _SCNetworkInterfaceCopyAll, referenced from:
      system_configuration::network_configuration::get_interfaces::hec77065b271b1561 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _SCNetworkInterfaceGetBSDName, referenced from:
      system_configuration::network_configuration::SCNetworkInterface::bsd_name::ha0ab64582943458e in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _SCNetworkInterfaceGetInterfaceType, referenced from:
      system_configuration::network_configuration::SCNetworkInterface::interface_type_string::h68e47ba8effc3241 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _SCNetworkInterfaceGetLocalizedDisplayName, referenced from:
      system_configuration::network_configuration::SCNetworkInterface::display_name::h45b785aea45c3017 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _SCNetworkReachabilityCreateWithAddress, referenced from:
      _$LT$system_configuration..network_reachability..SCNetworkReachability$u20$as$u20$core..convert..From$LT$core..net..socket_addr..SocketAddr$GT$$GT$::from::h6ac03fd0ef6ad317 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _SCNetworkReachabilityCreateWithAddressPair, referenced from:
      system_configuration::network_reachability::SCNetworkReachability::from_addr_pair::haa58ec16ccd384f0 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _SCNetworkReachabilityCreateWithName, referenced from:
      system_configuration::network_reachability::SCNetworkReachability::from_host::h5a862baa9924c1dd in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _SCNetworkReachabilityGetFlags, referenced from:
      system_configuration::network_reachability::SCNetworkReachability::reachability::h8efcd57427734e16 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _SCNetworkReachabilityScheduleWithRunLoop, referenced from:
      system_configuration::network_reachability::SCNetworkReachability::schedule_with_runloop::hf8e32742d4357ac4 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _SCNetworkReachabilityUnscheduleFromRunLoop, referenced from:
      system_configuration::network_reachability::SCNetworkReachability::unschedule_from_runloop::hb62dd8c9d7ce4a28 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _SCNetworkServiceCopyAll, referenced from:
      system_configuration::network_configuration::SCNetworkService::get_services::hfd924980bdaa6ebb in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _SCNetworkServiceGetEnabled, referenced from:
      system_configuration::network_configuration::SCNetworkService::enabled::hc5c7903eb5d232b6 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _SCNetworkServiceGetInterface, referenced from:
      system_configuration::network_configuration::SCNetworkService::network_interface::hc52379c06f994ff3 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _SCNetworkServiceGetServiceID, referenced from:
      system_configuration::network_configuration::SCNetworkService::id::h40c7459bebf1b5db in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _SCNetworkSetCopyCurrent, referenced from:
      system_configuration::network_configuration::SCNetworkSet::new::hf35f05c0ba06b9fc in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _SCNetworkSetGetServiceOrder, referenced from:
      system_configuration::network_configuration::SCNetworkSet::service_order::hd1c09af277b05e78 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _kSCDynamicStoreUseSessionKeys, referenced from:
      system_configuration::dynamic_store::SCDynamicStoreBuilder$LT$T$GT$::create_store_options::hc360a0be3ed310d3 in LDKNodeFFI[arm64][52](reqwest-5f2222f5565472a3.reqwest.eac4e8a186395011-cgu.0.rcgu.o)
  _kSCNetworkInterfaceType6to4, referenced from:
      system_configuration::network_configuration::SCNetworkInterfaceType::from_cfstring::haa23b0aca93a5944 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _kSCNetworkInterfaceTypeBluetooth, referenced from:
      system_configuration::network_configuration::SCNetworkInterfaceType::from_cfstring::haa23b0aca93a5944 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _kSCNetworkInterfaceTypeBond, referenced from:
      system_configuration::network_configuration::SCNetworkInterfaceType::from_cfstring::haa23b0aca93a5944 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _kSCNetworkInterfaceTypeBridge, referenced from:
      system_configuration::network_configuration::SCNetworkInterfaceType::from_cfstring::haa23b0aca93a5944 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _kSCNetworkInterfaceTypeEthernet, referenced from:
      system_configuration::network_configuration::SCNetworkInterfaceType::from_cfstring::haa23b0aca93a5944 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _kSCNetworkInterfaceTypeFireWire, referenced from:
      system_configuration::network_configuration::SCNetworkInterfaceType::from_cfstring::haa23b0aca93a5944 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _kSCNetworkInterfaceTypeIEEE80211, referenced from:
      system_configuration::network_configuration::SCNetworkInterfaceType::from_cfstring::haa23b0aca93a5944 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _kSCNetworkInterfaceTypeIPSec, referenced from:
      system_configuration::network_configuration::SCNetworkInterfaceType::from_cfstring::haa23b0aca93a5944 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _kSCNetworkInterfaceTypeIPv4, referenced from:
      system_configuration::network_configuration::SCNetworkInterfaceType::from_cfstring::haa23b0aca93a5944 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _kSCNetworkInterfaceTypeIrDA, referenced from:
      system_configuration::network_configuration::SCNetworkInterfaceType::from_cfstring::haa23b0aca93a5944 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _kSCNetworkInterfaceTypeL2TP, referenced from:
      system_configuration::network_configuration::SCNetworkInterfaceType::from_cfstring::haa23b0aca93a5944 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _kSCNetworkInterfaceTypeModem, referenced from:
      system_configuration::network_configuration::SCNetworkInterfaceType::from_cfstring::haa23b0aca93a5944 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _kSCNetworkInterfaceTypePPP, referenced from:
      system_configuration::network_configuration::SCNetworkInterfaceType::from_cfstring::haa23b0aca93a5944 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _kSCNetworkInterfaceTypePPTP, referenced from:
      system_configuration::network_configuration::SCNetworkInterfaceType::from_cfstring::haa23b0aca93a5944 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _kSCNetworkInterfaceTypeSerial, referenced from:
      system_configuration::network_configuration::SCNetworkInterfaceType::from_cfstring::haa23b0aca93a5944 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _kSCNetworkInterfaceTypeVLAN, referenced from:
      system_configuration::network_configuration::SCNetworkInterfaceType::from_cfstring::haa23b0aca93a5944 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _kSCNetworkInterfaceTypeWWAN, referenced from:
      system_configuration::network_configuration::SCNetworkInterfaceType::from_cfstring::haa23b0aca93a5944 in LDKNodeFFI[arm64][58](system_configuration-ede90d7c8f114b62.system_configuration.e6ca60e9865f4784-cgu.0.rcgu.o)
  _kSCPropNetProxiesHTTPEnable, referenced from:
      core::ops::function::FnOnce::call_once::h09558ef78fe7094a in LDKNodeFFI[arm64][52](reqwest-5f2222f5565472a3.reqwest.eac4e8a186395011-cgu.0.rcgu.o)
  _kSCPropNetProxiesHTTPPort, referenced from:
      core::ops::function::FnOnce::call_once::h09558ef78fe7094a in LDKNodeFFI[arm64][52](reqwest-5f2222f5565472a3.reqwest.eac4e8a186395011-cgu.0.rcgu.o)
  _kSCPropNetProxiesHTTPProxy, referenced from:
      core::ops::function::FnOnce::call_once::h09558ef78fe7094a in LDKNodeFFI[arm64][52](reqwest-5f2222f5565472a3.reqwest.eac4e8a186395011-cgu.0.rcgu.o)
  _kSCPropNetProxiesHTTPSEnable, referenced from:
      core::ops::function::FnOnce::call_once::h09558ef78fe7094a in LDKNodeFFI[arm64][52](reqwest-5f2222f5565472a3.reqwest.eac4e8a186395011-cgu.0.rcgu.o)
  _kSCPropNetProxiesHTTPSPort, referenced from:
      core::ops::function::FnOnce::call_once::h09558ef78fe7094a in LDKNodeFFI[arm64][52](reqwest-5f2222f5565472a3.reqwest.eac4e8a186395011-cgu.0.rcgu.o)
  _kSCPropNetProxiesHTTPSProxy, referenced from:
      core::ops::function::FnOnce::call_once::h09558ef78fe7094a in LDKNodeFFI[arm64][52](reqwest-5f2222f5565472a3.reqwest.eac4e8a186395011-cgu.0.rcgu.o)
clang: error: linker command failed with exit code 1 (use -v to see invocation)

** BUILD FAILED **
```

It seems this was already broken for the 0.2 release. Currently investigating what causes the linker to fail.

### LINKED ISSUE #279: Check if we also need to apply `Info.plist` fixes
Apple is fun. We should see if we need to fix this, too: https://github.com/bitcoindevkit/bdk-ffi/pull/466

### PR DESCRIPTION
**Pull request by tnull**
Fixes #231 
Fixes #279 

Unfortunately Apple broke our Swift packages with the recent Xcode 15.3 release. Here, we add the (now) required information to (hopefully) make our builds works again. 

Moreover, we add a oneliner, automatically patching `LDKNode.swift` with the fix for #231 in the build script.

### CODE REVIEW

**Comment by reez** [comment]
**tl;dr I got this working in Xcode 15.4... but since what we are fixing is a bit of a black box here's my running notes on the process of how I tested and the fix I applied in case anyone else wants to run thru what I did or did something different or has any additional insights**

### Environment

I have the new Xcode 15.4, as well as still have the old (pre-everything-breaks) Xcode 15.2.

I’m using Xcode 15.2 as sanity tests for where everything should be working regardless so I can compare vs fixes for Xcode 15.4.

### Versions 

I’m using a version of Monday https://github.com/reez/Monday/pull/88/commits that is keeping up with all of the new changes in ldk-node https://github.com/lightningdevkit/ldk-node/commits/main/ … 

the last change on ldk-node is b7c4862221a5f78b099293282ceabc4c13d41f94 which matches the local build of the swift bindings I’m using for that branch of Monday https://github.com/reez/Monday/pull/88/commits/81581bd3e4c667eb870cb5a17b853fad194f7ff5

### Current PR

https://github.com/tnull/ldk-node/tree/2024-05-fix-swift-builds only adds two commits on top of b7c4862221a5f78b099293282ceabc4c13d41f94 

When I pull down https://github.com/lightningdevkit/ldk-node/pull/294 to use in Monday as a remote package for https://github.com/reez/Monday/pull/88 I get a bunch of errors (as if all of the new commits in ldk-node from the past month+ or so haven’t been added) ...

Example: `NodeStatus` “Cannot find type ‘NodeStatus’ in scope”, even though I have the code for it https://github.com/reez/Monday/pull/88/files#diff-1f8112828f9e05100c7ec541d3485329e2f6ce3c20e634e2998cda39c41e6afeR18 , which I guess makes sense if https://github.com/tnull/ldk-node/blob/2024-05-fix-swift-builds/bindings/swift/Sources/LDKNode/LDKNode.swift hasn’t been updated in 4 months (even though code for `NodeStatus` has been merged into main). 

Screenshot 2024-05-20 at 10 46 31 AM

### Local Build

So lets locally build https://github.com/tnull/ldk-node/tree/2024-05-fix-swift-builds

I’ll fork tnull repo and build swift bindings locally.

I got these bindings to build and run successfully on Xcode 15.2!

Screenshot 2024-05-20 at 11 06 00 AM

And   I got these bindings to build and run successfully on Xcode 15.4! 

Screenshot 2024-05-20 at 11 08 18 AM

Now trying to archive Monday and upload to App Store Connect (last TestFlight build was 91), I got a failure for `Info.plist` (I ran into a small extra nuance like this when trying to do the same thing in bdk-ffi/bdk-swift when actually archiving+uploading with BDKSwiftExample Wallet so this might be what I ran into with BDK as well, lets do a BDK+LDKNode PR review/comparison real quick to see if we can catch anything)

Screenshot 2024-05-20 at 11 12 57 AM

### This PR vs BDK PR (It looked like part of the Info plists were different than mine)

#### bindings/swift/LDKNodeFFI.xcframework/Info.plist  

https://github.com/lightningdevkit/ldk-node/pull/294/files#diff-b50a77f58d4ab17a37b3151ffd1647c1c542fd693541edc82271d2a684556397R50-R51 `100.0`

https://github.com/bitcoindevkit/bdk-ffi/pull/466/files#diff-1d35206bd681a5c7562b5984ff5b2ce4cc413701e8c7195d5dcd614e653965b6R50-R51 `15.0`

#### bindings/swift/LDKNodeFFI.xcframework/ios-arm64/LDKNodeFFI.framework/Info.plist

https://github.com/lightningdevkit/ldk-node/pull/294/files#diff-62d9da0702a87267b9ac959c6b8e9262a604cbaa01412cc7e5567a0e9f4c007cR15-R16 `15.0`

https://github.com/bitcoindevkit/bdk-ffi/pull/466/files#diff-0006e59b40dfcfac3a5ebe4d63e74f75154f415f45b7e95f0dc8318aa5a604a0R15-R16 `100`

#### bdk-swift/bdkFFI.xcframework/ios-arm64_x86_64-simulator/bdkFFI.framework/Info.plist

Looks good

#### bindings/swift/LDKNodeFFI.xcframework/macos-arm64_x86_64/LDKNodeFFI.framework/Info.plist

Looks good

#### scripts/uniffi_bindgen_generate_swift.sh

  We dont have this change in bdk-ffi, I’m just going to assume these are strictly for #231 

#### Other

Also just make sure versions match Package.swift of specific repo (that's what I'm trying to match in the Info.plist files)

Yes both ldk-node https://github.com/tnull/ldk-node/blob/2024-05-fix-swift-builds/bindings/swift/Package.swift and bdk-ffi https://github.com/bitcoindevkit/bdk-ffi/blob/release/0.31/bdk-swift/Package.swift seem to support the same: 
- .macOS(.v12)
- .iOS(.v15)

### My additional fix

So lets update tnull fork branch with my additional fixes applied on top

Screenshot 2024-05-20 at 11 27 17 AM
Screenshot 2024-05-20 at 11 27 20 AM

Build succeeds still!

Screenshot 2024-05-20 at 11 29 03 AM

I was able to archive and upload to App Store Connect with Xcode 15.4!

Screenshot 2024-05-20 at 11 29 38 AM

I also tested using that new TestFlight build 92 on a physical device myself as well, and that worked!

Everything works similar to bdk now where at least from my testing I'm able to do all of these things in Xcode 15.4:
- Build
- Run
- Archive
- Upload to App Store Connect

### Follow up:

All of the changes that make this work locally just need to propagate to `LDKNode.swift` on the remote repo so that this will work for remotely pulling this into a project. Happy to help look into that, and also test that that works properly for Xcode 15.4 too, but I think that all should be pretty simple and quick to do (famous last words).

File: bindings/swift/LDKNodeFFI.xcframework/Info.plist [hunk 1]
```diff
@@ -43,6 +47,8 @@
 			</array>
 			<key>SupportedPlatform</key>
 			<string>ios</string>
+			<key>MinimumOSVersion</key>
+			<string>100.0</string>
```
Comment (reez) [inline bindings/swift/LDKNodeFFI.xcframework/Info.plist:51]: I think this should be `15.0`

File: bindings/swift/LDKNodeFFI.xcframework/ios-arm64/LDKNodeFFI.framework/Info.plist [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+	<key>CFBundleName</key>
+	<string>LDKNodeFFI</string>
+	<key>CFBundleVersion</key>
+	<string>0.3.0</string>
+	<key>CFBundleShortVersionString</key>
+	<string>0.3.0</string>
+	<key>CFBundleExecutable</key>
+	<string>LDKNodeFFI</string>
+	<key>MinimumOSVersion</key>
+	<string>15.0</string>
```
Comment (reez) [inline bindings/swift/LDKNodeFFI.xcframework/ios-arm64/LDKNodeFFI.framework/Info.plist:16]: I think this should be `100`

**Comment by reez** [comment]
> > Example: `NodeStatus` “Cannot find type ‘NodeStatus’ in scope”, even though I have the code for it https://github.com/reez/Monday/pull/88/files#diff-1f8112828f9e05100c7ec541d3485329e2f6ce3c20e634e2998cda39c41e6afeR18 , which I guess makes sense if https://github.com/tnull/ldk-node/blob/2024-05-fix-swift-builds/bindings/swift/Sources/LDKNode/LDKNode.swift hasn’t been updated in 4 months (even though code for `NodeStatus` has been merged into main).
> 
> Right, this is expected. We only update the checked-in files and the `xcframework.zip` upon the new release. IIUC, you could successfully use a local build to build and publish. I just amended the minor oversight (regarding `100.0`) above.
> 
> So, if there's nothing else from your side, I'll merge this PR tomorrow. Then I'll backport it on top of 0.2.1 will release a fixed v0.2.2.

Ok sounds good, then yeah all my testing is done now until it gets released, and then I'll just make sure to test again once the new release is cut just for good measure. 🚀

**Comment by reez** [comment]
> @reez Just backported these fixes and released v0.2.2 via #296. Let me know if it works for you as expected!

Everything works as expected from me pulling in v0.2.2 remotely, build+run+archive+upload 🚀 https://github.com/reez/Monday/pull/89
